src/main/kotlin/com/reduks/reduks/combine_reducers/CombineReducers.kt

```diff
@@ -0,0 +1,21 @@
+package com.reduks.reduks.combine_reducers
+
+import com.reduks.reduks.Action
+
+/**
+ * Created by bloder on 18/02/18.
+ */
+
+class CombineReducers<State>(private val reducers: Array<out (State, Action<State>) -> State>) {
+
+    fun combine() : (State, Action<State>) -> State = fun (initialState: State, action: Action<State>) : State {
+        var state: State = initialState
+        reducers.forEach {
+            val currentState = it(state, action)
+            state = if (currentState != state) currentState else state
+        }
+        return state
+    }
+}
+
+fun <State> Any.combineReducers(vararg reducers: (State, Action<State>) -> State) : (State, Action<State>) -> State = CombineReducers(reducers).combine()
```

Comment:
This is useless IMHO

Reply:
It can be a simple constructor.

Reply:
I like to use extension here to look like redux 

Redux
```javascript
combineReducers({ reducer1,  reducer2 })
```

Reduks
```javascript
combineReducers(reducer1, reducer2)
```

We need to think about library design, could it be a simple constructor? yes, but how we discussed before, it's interesting we design the library like redux did